src/Files.App.CsWin32/Extras.cs

```diff
@@ -44,6 +44,25 @@ public static unsafe nint SetWindowLongPtr(HWND hWnd, WINDOW_LONG_PTR_INDEX nInd
 		public const int PixelFormat32bppARGB = 2498570;
 	}
 
+	namespace Foundation
+	{
+		public partial struct PCSTR
+		{
+			public static unsafe PCSTR FromString(string value)
+			{
+				fixed (byte* p = global::System.Text.Encoding.UTF8.GetBytes(value))
+				{
+					return new PCSTR(p);
```

Comment:
I have no idea what you doing. Once getting out of `fixed`, the `p` won't be guaranteed to be fixed any longer.

Reply:
How do I then construct the struct from a string/int value?

Reply:
You have to pin the address unless the method you want to call is returned.

```
fixed (char* pName = name)
{
    // PInvoke.Method(..., pName, ...);
}
```

So, this case you can't create a helper like that.